### PR TITLE
Don't convert valtree to constvalue during normalization

### PR DESCRIPTION
**Pull request by b-naber**
r? @lcnr

### CODE REVIEW

File: compiler/rustc_trait_selection/src/traits/query/normalize.rs [hunk 1]
```diff
@@ -353,18 +353,8 @@ impl<'cx, 'tcx> FallibleTypeFolder<'tcx> for QueryNormalizer<'cx, 'tcx> {
     ) -> Result<mir::ConstantKind<'tcx>, Self::Error> {
         Ok(match constant {
             mir::ConstantKind::Ty(c) => {
```
Comment (lcnr) [inline compiler/rustc_trait_selection/src/traits/query/normalize.rs:355]: now that's just the default impl, isn't it? :grin: 

don't think you have to overwrite this method at all now

Comment (b-naber) [reply compiler/rustc_trait_selection/src/traits/query/normalize.rs:355]: ^^

Comment (b-naber) [reply compiler/rustc_trait_selection/src/traits/query/normalize.rs]: eh actually it isnt ^^: https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_middle/ty/fold.rs.html#193

Comment (lcnr) [reply compiler/rustc_trait_selection/src/traits/query/normalize.rs:355]: ah, then just `c.try_super_fold_with` i guess :grin:

**Comment by lcnr** [comment]
with acceptable perf, r=me

@bors delegate+

**Pull request merged**